cli/cmd/inject_test.go

```diff
@@ -34,19 +34,21 @@ func testUninjectAndInject(t *testing.T, tc injectYAML) {
 	}
 
 	actualOutput := output.String()
-	expectedOutput := readOptionalTestFile(t, tc.goldenFileName)
-	if expectedOutput != actualOutput {
-		t.Errorf("Result mismatch.\nExpected: %s\nActual: %s", expectedOutput, actualOutput)
+	expectedOutput := readTestdata(t, tc.goldenFileName)
+	if actualOutput != expectedOutput {
+		writeTestdataIfUpdate(t, tc.goldenFileName, output.Bytes())
+		diffCompare(t, expectedOutput, actualOutput)
```

Comment:
Method signature for `diffCompare` is:

```
func diffCompare(t *testing.T, actual string, expected string) {
```

So this should be
```suggestion
		diffCompare(t, actualOutput, expectedOutput)
```

In order for the diff to be colored correctly. Same goes for the rest of the `diffCompare` calls in this file.